src/resource.js

```diff
@@ -1,6 +1,58 @@
 'use strict'
 
-export default () => {
+import url from 'url'
+import jts from 'jsontableschema'
+import _ from 'lodash'
 
-}
+export default class Resource {
 
+  constructor(descriptor) {
+    this._descriptor = descriptor
+    this.REMOTE_PROTOCOLS = ['http:', 'https:', 'ftp:', 'ftps:']
+  }
+
+  getDescriptor() {
+    return this._descriptor
+  }
+
+  getName() {
+    return this.getDescriptor()['name']
+  }
+
+  getType() {
+    const resourceField = this._getResourceField()
+
+    if (resourceField === 'data') {
+      return 'inline'
+    }
+
+    const protocol = url.parse(this.getDescriptor()[resourceField])['protocol']
+    if (_.includes(this.REMOTE_PROTOCOLS, protocol)) {
+      return 'remote'
+    } else {
+      return 'local'
+    }
+
+  }
+
+  getSource() {
+    return this.getDescriptor()[this._getResourceField()]
+  }
+
+  getTable() {
```

Comment:
No. It's I've missed that `jst.Table()` is a Promise :smiley: 

So what' our options? I think we still could return promise (instantiating `this._table` in constructor) for usage  like this:

```javascript
table = await resource.table  // return table or null
if (table) {
  // use Table API
}
```
```javascript
resource.table.then(table => {
  if (table) {
    // use Table API
  }
})
```

cc @zhenyab